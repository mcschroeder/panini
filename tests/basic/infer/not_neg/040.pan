not : (x:bool) -> {y:bool|y = ~x}

f40 : bool -> {y:bool|?}
f40 = \x:bool. not x

f41 : {x:bool|?} -> {y:bool|?}
f41 = \x:bool. not x

