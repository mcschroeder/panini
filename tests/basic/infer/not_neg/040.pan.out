f40 : 𝔹 → {y:𝔹 | true}
f41 : {x:𝔹 | true} → {y:𝔹 | y = ¬x}